### PR TITLE
fix: update Update MWAAEnviroment to create VPC and MWAAExecutionRole with resource prefix

### PR DESCRIPTION
**Pull request by devansh-gandhi**
Description

Currently,ddk fails to deploy multiple MWAAEnvironment as VPC and MWAAExecutionRole have fixed ids.

Changes Made
VPC and MWAAExecutionRole have resource_prefox / props.name added to their id to ensure new resources are created for each MWAAEnvironment. 

Related Issue
#438

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 3e54a5041b7cd4e93eeb4c1f39613bf8d2b376a4
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=070e366d-55ce-442f-a8f3-a037b9f0a9dd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/core/mwaa-environment.ts [hunk 1]
```diff
@@ -265,7 +265,7 @@ export class MWAAEnvironment extends Construct {
       throw new Error("Vpc Cidr Range must of size >=16 and <=20");
     }
     const subnetCIDRMask = vpcCIDRMask + 4;
-    const vpc = new ec2.Vpc(scope, "Vpc", {
+    const vpc = new ec2.Vpc(scope, `${resourceName} Vpc`, {
```
Comment (malachi-constant) [inline src/core/mwaa-environment.ts:268]: This works for me

File: src/core/mwaa-environment.ts [hunk 2]
```diff
@@ -144,7 +144,7 @@ export class MWAAEnvironment extends Construct {
       }
     }
 
-    const mwaaExecutionRole = new iam.Role(scope, "MWAA Execution Role", {
+    const mwaaExecutionRole = new iam.Role(scope, `${props.name} MWAA Execution Role`, {
```
Comment (malachi-constant) [inline src/core/mwaa-environment.ts:147]: This pattern will need to be followed for all resources, (i.e. `s3deploy.BukcketDeployment()`, etc.

**Review by malachi-constant — CHANGES_REQUESTED** [review]
I'd like to see a test deploying two instances before we can merge this. Some additional  will need `props.name` prepended to the construct id as well

File: src/core/mwaa-environment.ts (hunk 2 above)
Comment (devansh-gandhi) [reply src/core/mwaa-environment.ts:147]: Hi, I have added the prefix for all resources in the stack. Regarding the vpc name, you mentioned it was okay i used the resourceName that has -mwaa at the end, but let me know if i should revert to using props.name directly from the function args to make it same accross all resources. 

I have also created a test in the mwaa-environment.test.ts that check if there are 2 MWAAEnvironments created. 

Please let me know if any additional modifications or tests are required.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 2a8eb9e4112954a186e61ca8396a7441860cf468
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ec88f74d-d149-4c8c-850b-0cba7f3faa61%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

**Pull request merged**